### PR TITLE
Expose SliderBody's snaking and border colour

### PR DESCRIPTION
**Pull request by smoogipoo**

### CODE REVIEW

File: osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSlider.cs [hunk 1]
```diff
@@ -87,6 +89,13 @@ public DrawableSlider(Slider s)
             }
         }
 
+        [BackgroundDependencyLoader]
+        private void load(OsuConfigManager config)
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSlider.cs]: why was this moved here?

File: osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBody.cs [hunk 2]
```diff
@@ -51,6 +53,24 @@ public Color4 AccentColour
...
+        {
+            get { return borderColour; }
+            set
+            {
+                if (borderColour == value)
+                    return;
+                borderColour = value;
+
+                if (LoadState == LoadState.Ready)
+                    Schedule(reloadTexture);
```
Comment (peppy) [inline osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBody.cs]: don't think the schedule here is required. we shouldn't be doing cross-thread border colour changes post-load.

**Review by peppy — CHANGES_REQUESTED** [review]

File: osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBody.cs (hunk 2 above)
Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/Objects/Drawables/Pieces/SliderBody.cs]: This is the same thing as what `AccentColour` does. This is probably done so we don't reload the texture multiple times prior to load() being run (which does `reloadTexture` itself).

File: osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSlider.cs (hunk 1 above)
Comment (smoogipoo) [reply osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSlider.cs]: Because it shouldn't be done at such a low level like `SliderBody` - something that is exposed publicly for the purpose of drawing a slider body. We may want to use `SliderBody` without having an entire hit object.

**Review by peppy — APPROVED** [review]
